test/framework/k8s_utils.go

```diff
@@ -371,6 +373,24 @@ func DownloadKubeconfig(ctx context.Context, client kubernetes.Interface, namesp
 	return nil
 }
 
+// DownloadAdminKubeconfigForShoot requests an admin kubeconfig for the given shoot and writes the kubeconfig to the
+// given download path. The kubeconfig expires in 6 hours.
+func DownloadAdminKubeconfigForShoot(ctx context.Context, client kubernetes.Interface, shoot *gardencorev1beta1.Shoot, downloadPath string) error {
+	const expirationSeconds = 6 * 3600 // 6h
+	kubeconfig, err := access.RequestAdminKubeconfigForShoot(ctx, client, shoot, pointer.Int64(expirationSeconds))
+	if err != nil {
+		return err
+	}
+
+	if downloadPath != "" {
```

Comment:
🤔 If `downloadPath` is empty, the requested kubeconfig would not be accessible out of this function. I'm not sure, if there's a scenario, where this could happen though. But maybe there is usage of the `ShootCreationFramework` for tests that run outside of testmachinery.

Maybe the check could be run before actually requesting a kubeconfig and if it's empty, log some info message?

Reply:
Makes sense. I now moved the check for the `downloadPath` to the caller.